pkgs/applications/version-management/anu/default.nix

```
@@ -0,0 +1,44 @@
+{ stdenv
+, fetchCrate
+, rustPlatform
+, pkg-config
+, clang
+, libclang
+, libsodium
+, openssl
+, xxHash
+, zstd
+, CoreServices
+, Security
```

These are darwin-specific; might be best to leave them as optional? (Not really sure how this is normally handled, first time I have seen this platform specific dependencies)

```suggestion
, CoreServices ? null
, Security ? null
```

---

Isn't `stdenv.lib.optionals stdenv.isDarwin [ CoreServices Security ])` sufficient? There are some similar instances in nixpkgs. I'd like to hear from experts whether this is idiomatic or not too, though.

---

I tried building it on NixOS and had an issue of insufficient arguments. I also don't know how it is normally handled. :C

---

Ah, I see. Thank you for sharing your situation.

---

Strange, I built it on NixOS just fine without that...

---

20.09?

---

As reference, I am on NixOS 20.09 with `nix-unstable` as the user channel. I used `nix-build -E "with import <nixpkgs> {}; callPackage ./anu.nix { libclang = llvmPackages.libclang; }"` to build this expression and that required the suggested changes.

---

> I'd like to hear from experts whether this is idiomatic or not too, though.

I don't think this is idiomatic, but I'm definitely not an expert.

---

Indeed, passing dependencies in regardless of the platform is the idiomatic approach. We don't reference platform specific dependencies with pkgs.hello instead of passing them in as arguments either.